src/common/components/elements/CurrentListings.tsx

```diff
@@ -141,14 +140,15 @@ function reducer(state: DiscoveryToolState, action: DiscoveryToolAction): Discov
       };
     case 'LOAD_MORE_LISTINGS':
       const newCursor = state.cursor + pageSize;
+      const listings = new Set([
+        ...state.listingsOnDisplay,
+        ...state.filteredAndSortedListings.slice(state.cursor, newCursor),
+      ])
```

Comment:
@dandlezzz I'm not sure this can fix the root cause. new Set() only dedupes if the array is a list of primitives, not objects.

I created a codesandbox to verify: https://codesandbox.io/s/lingering-thunder-pwqmh?file=/src/App.tsx

I'll do some more tests on my end

Reply:
Turns out in this case it actually will work :O https://stackoverflow.com/questions/39997067/es6-unique-array-of-objects-with-set 

Because the object references are the same.

However, I think this has something to do with the cursor, so I'll look into that

Reply:
Solved it! Turns out I was resetting the cursor to 0 instead of page size on sort and filter. Sill!